streaming/src/main/java/org/openmrs/analytics/FhirEventWorker.java

```diff
@@ -92,15 +51,15 @@ public void process(Event event) {
 				log.info("Skipping non-FHIR event " + event);
 				return;
 			}
+			
 			log.info("FHIR resource URL is: " + fhirUrl);
-			String fhirJson = fetchFhirResource(feedBaseUrl + fhirUrl);
-			log.info("Fetched FHIR resource: " + fhirJson);
-			// Creating this resource is not really needed for the current purpose as we can simply
-			// send the JSON payload to GCP FHIR store. This is kept for demonstration purposes.
-			T resource = parserFhirJson(fhirJson);
+			T resource = (T) openmrsUtil.fetchFhirResource(fhirUrl);
 			String resourceId = resource.getIdElement().getIdPart();
-			log.info(String.format("Parsed FHIR resource ID is %s and IdBase is %s", resourceId, resource.getIdBase()));
-			fhirStoreUtil.uploadResourceToCloud(this.resourceType, resourceId, fhirJson);
```

Comment:
The idea here was to prevent one parsing of the JSON payload (and then deserializing it again when sending to the sink FHIR store). I agree that in most cases we would want to parse the resource too but I just want to make sure you are aware of the unnecessary performance overhead.

Reply:
I'm trying to think out if this is possible with the HAPI client - `client.read()` returns a Resource object, so it automatically parses the JSON payload. This resource needs to be serialized before sending to the sink, so it's not the best option in terms of performance. The faster approach would be to just pass along the JSON without touching it, like you implemented originally.

Reply:
Right, this mostly impacts the use case where the sink is another FHIR store, so I leave it to you to decide and/or do some performance evaluation. For the use case where the sink is a data warehouse (or some non-JSON files) we need to do some transformation hence converting to Resource is probably necessary; marking as resolved.